### PR TITLE
fix: inject KubeVersion and ApiVersions in Chartify rendering

### LINKED ISSUE #1623: `--args` parameters doesn't work along with a kustomize patch
### Operating system

MacOS sonoma

### Helmfile Version

v0.166.0

### Helm Version

v3.15.2

### Bug description

When using a `strategicMergePatches`, the arguments (`--args`) passed through the helmfile CLI are not passed for the first `helm template` rendering, they are used only on the latest `helm template` rendering that occurs on the kustomize output (`all.patched.yaml`) but for example an `--api-versions` override here is useless...

See the following debug output:
```
running helm template --debug=false --output-dir=/var/folders/kf/vwr6x12s6j1_6v0j44v300800000gn/T/chartify951191064/foo/helmx.1.rendered foo /var/folders/kf/vwr6x12s6j1_6v0j44v300800000gn/T/chartify951191064/foo

running kustomize build /var/folders/kf/vwr6x12s6j1_6v0j44v300800000gn/T/chartify951191064/foo --output /var/folders/kf/vwr6x12s6j1_6v0j44v300800000gn/T/chartify951191064/foo/all.patched.yaml --enable-alpha-plugins

exec: helm template foo /var/folders/kf/vwr6x12s6j1_6v0j44v300800000gn/T/chartify951191064/foo --api-versions=v1/myApi --debug
```

### Example helmfile.yaml

```yaml
---
releases:
  - name: foo
    chart: ../chart
    strategicMergePatches:
      - apiVersion: policy/v1
        kind: PodDisruptionBudget
        metadata:
          name: foo
          namespace: bar
          annotations:
            patch: ok
```

`helmfile template . --args --api-versions="v1/myApi"`

### Error message you've seen (if any)

overriding an api-versions through the CLI is not take into account when used along a kustomize patch

### Steps to reproduce

https://github.com/LVD-Consulting/helmfile-bug-args

### Working Helmfile Version

none

### Relevant discussion

_No response_

### PR DESCRIPTION
**Pull request by xabufr**
fixes #1623

### CODE REVIEW

File: pkg/state/state.go [hunk 1]
```diff
@@ -2822,24 +2823,32 @@ func (st *HelmState) appendValuesControlModeFlag(flags []string, reuseValues boo
...
-		flags = append(flags, "--kube-version", r.KubeVersion)
+		return r.KubeVersion
 	case st.KubeVersion != "":
-		flags = append(flags, "--kube-version", st.KubeVersion)
+		return st.KubeVersion
+	}
+	return ""
+}
+
+func (st *HelmState) appendApiVersionsFlags(flags []string, r *ReleaseSpec, kubeVersion string) []string {
```
Comment (yxxhero) [inline pkg/state/state.go:2845]: add some unittests for this function. please.

Comment (yxxhero) [reply pkg/state/state.go:2845]: @xabufr

Comment (xabufr) [reply pkg/state/state.go:2845]: @yxxhero I've added test for `appendApiVersionsFlags`, is it ok for you?

Comment (yxxhero) [reply pkg/state/state.go:2845]: LGTM, but why do this PR fix https://github.com/helmfile/helmfile/issues/1623? @xabufr

Comment (xabufr) [reply pkg/state/state.go:2845]: ApiVersion and KubeVersion wasn't correctly injected in Chartify [here](https://github.com/helmfile/helmfile/pull/1624/files#diff-7e4e87a1e777423a3d9a915e00fb78a9c4fc2f4c476863f60b67e89b9c635697R1259), with this PR I keep the same behavior than without chartify: use release or state versions depending what is specified

**Pull request merged**